src/meta/timing.rs

```diff
@@ -145,51 +159,63 @@ impl TimedCode {
 
     /// Start the next measurement
     pub fn start(&mut self) {
-        if cfg!(test) {
+        if cfg!(test) || cfg!(not(feature = "timing")) {
             return;
         }
 
         debug_assert!(self.info.start_thread.is_none());
 
-        self.info.start_system = Some(HighResolutionClock::now());
-        self.info.start_process = Some(ProcessRealCPUClock::now());
-        self.info.start_thread = Some(ThreadClock::now());
+        #[cfg(feature = "timing")]
+        {
+            self.info.start_system = Some(HighResolutionClock::now());
+            self.info.start_process = Some(ProcessRealCPUClock::now());
+            self.info.start_thread = Some(ThreadClock::now());
+        }
     }
 
     /// Stop the current measurement and save the times
     pub fn stop(&mut self) -> Duration {
-        if cfg!(test) {
+        if cfg!(test) || cfg!(not(feature = "timing")) {
             return Duration::ZERO;
         }
 
-        debug_assert!(self.info.start_system.is_some());
-
-        let start_system = self
-            .info
-            .start_system
-            .expect("start() must be called before calling stop()");
-        let start_process = self
-            .info
-            .start_process
-            .expect("start() must be called before calling stop()");
-        let start_thread = self
-            .info
-            .start_thread
-            .expect("start() must be called before calling stop()");
-
-        let duration_system = HighResolutionClock::now() - start_system;
-        let duration_process = ProcessRealCPUClock::now() - start_process;
-        let duration_thread = ThreadClock::now() - start_thread;
-
-        self.info.total_system_time += duration_system;
-        self.info.total_process_time += duration_process;
-        self.info.total_thread_time += duration_thread;
-        self.info.start_system = None;
-        self.info.start_process = None;
-        self.info.start_thread = None;
-        self.info.runs += 1;
-
-        duration_thread
+        #[cfg(not(feature = "timing"))]
+        {
+            // This is not reachable, but the function would otherwise need a return value
+            panic!();
```

Comment:
Ah, I see, it is not unreachable, but won't be called if timing is inactive. In this case I would just propose to make the comment the error message of `panic!`.
Also saying it is not reachable is sort of a lie, because it depends on the outside code. Maybe state that the operation is not supported?